Obsidian/Server.cs

```diff
@@ -559,7 +597,18 @@ private async Task ServerLoop()
 
             short itersPerSecond = 0;
             var stopWatch = Stopwatch.StartNew(); // for TPS measuring
-
+            _ = Task.Run(() =>
+            {
+                while (true)
+                {
+                    if (stopWatch.ElapsedMilliseconds >= 1000L)
```

Comment:
I'm not sure how good idea this is. My proposal would be to replace original code with something like
```cs
TPS = (short)(1.0 / stopwatch.Elapsed.TotalSeconds);
stopwatch.Restart();
```
which would remove the need of tracking how many iterations happened in a second (`itersPerSecond`) and would update the `TPS` value every tick, instead of every second.